controllers/modals/submission/ViewSubmissionMetadataHandler.inc.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * @file controllers/modals/submission/ViewSubmissionMetadataHandler.inc.php
+ *
+ * Copyright (c) 2014-2019 Simon Fraser University
+ * Copyright (c) 2003-2019 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ViewSubmissionMetadataHandler
+ * @ingroup controllers_modals_viewSubmissionMetadataHandler
+ *
+ * @brief Display submission metadata.
+ */
+
+// Import the base Handler.
+import('classes.handler.Handler');
+
+class ViewSubmissionMetadataHandler extends handler {
+
+	/**
+	 * Constructor.
+	 */
+	function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(array(ROLE_ID_REVIEWER), array('display'));
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.SubmissionAccessPolicy');
+		$this->addPolicy(new SubmissionAccessPolicy($request, $args, $roleAssignments));
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * Display metadata
+	 */
+	function display($args, $request) {
+		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		$reviewAssignment = $this->getAuthorizedContextObject(ASSOC_TYPE_REVIEW_ASSIGNMENT);
+		$templateMgr = TemplateManager::getManager($request);
+
+		if ($reviewAssignment->getReviewMethod() != SUBMISSION_REVIEW_METHOD_DOUBLEBLIND) { /* SUBMISSION_REVIEW_METHOD_BLIND or _OPEN */
+			$templateMgr->assign('authors', $submission->getAuthorString());
+		}
+
+		$templateMgr->assign(array(
+			'title' => $submission->getLocalizedFullTitle(),
+			'abstract' => $submission->getLocalizedAbstract(),
+		));
+
+		$publication = $submission->getCurrentPublication();
+		$submissionKeywordDao = DAORegistry::getDAO('SubmissionKeywordDAO');
+		$submissionSubjectDao = DAORegistry::getDAO('SubmissionSubjectDAO');
+		$submissionDisciplineDao = DAORegistry::getDAO('SubmissionDisciplineDAO');
+		$submissionAgencyDao = DAORegistry::getDAO('SubmissionAgencyDAO');
+		$submissionLanguageDao = DAORegistry::getDAO('SubmissionLanguageDAO');
```

Comment:
The controlled vocab is available directly from the `publication`, rather than going through the DAO. So you can do the following:

```
$keywords = $publication->getData('keywords');
```

Reply:
check